telegram/ext/dispatcher.py

```diff
@@ -247,78 +247,51 @@ def _pooled(self):
-        """Queue a function (with given args/kwargs) to be run asynchronously.
+            # If we arrive here, an exception happened in the promise and was neither
+            # DispatcherHandlerStop nor raised by an error handler. So we can and must handle it
+            try:
+                self.dispatch_error(promise.update, promise.exception, promise=promise)
+            except Exception:
+                self.logger.exception('An uncaught error was raised while handling the error.')
+
+    def run_async(self, func, *args, update=None, error_handling=True, **kwargs):
+        """Queue a function (with given args/kwargs) to be run asynchronously. Exceptions raised
+        by the asynchronously run function will be handled by the error handlers registered with
+        :meth:`add_error_handler`.
 
         Warning:
             * If you're using ``@run_async``/:meth:`run_async` you cannot rely on adding custom
               attributes to :class:`telegram.ext.CallbackContext`. See its docs for more info.
-            * When using :meth:`run_async` within an asynchronously run error handler (see
-              :meth:`add_error_handler`) and ``update`` is passed, you *must* also pass
-              ``error_handler``, as an exception within ``func`` would cause an infinite loop
-              otherwise.
-            * passing ``error_handler`` is not supported for the old style API, i.e.
-              ``use_context=False``
+            * When using :meth:`run_async` within an error handler (see :meth:`add_error_handler`),
+              you *must* also set :attr:`error_handling` to :obj:`False`, as an exception within
+              :attr:`func` would cause an infinite loop otherwise. Eventual exceptions will then
+              only be logged.
 
         Args:
             func (:obj:`callable`): The function to run in the thread.
             *args (:obj:`tuple`, optional): Arguments to ``func``.
             update (:class:`telegram.Update`, optional): The update associated with the functions
-                call.
-            error_handler (:obj:`callable`, optional): An error handler to call in case of an
-                exception during the call of ``func``. The callback signature of ``error_handler``
-                must be
-
-                ``def error_handler(context: CallbackContext, *args, **kwargs)``
-
-                or
-
-                ``def error_handler(update: Update, context: CallbackContext, *args, **kwargs)``
-
-                if ``update`` is passed, where ``*args`` and ``**kwargs`` are the arguments for
-                ``func``.
+                call. If passed, it will be available in the error handlers, in case an exception
+                is raised by :attr:`func`.
+            error_handling (:obj:`bool`, optional): Whether exceptions raised by :attr:`func`
+                may be handled by error handlers. Defaults to :obj:`True`.
```

Comment:
See my comment above regarding hiding `error_handling` implementation detail from the user.